src/services/codefixes/fixReturnTypeInAsyncFunction.ts

```diff
@@ -39,14 +39,20 @@ namespace ts.codefix {
     });
 
     function getInfo(sourceFile: SourceFile, checker: TypeChecker, pos: number): Info | undefined {
-        const returnTypeNode = getReturnTypeNode(sourceFile, pos);
+        if (isInJSFile(sourceFile)) {
+            return undefined;
+        }
+
+        const token = getTokenAtPosition(sourceFile, pos);
+        const func = findAncestor(token, isFunctionLikeDeclaration);
+        const returnTypeNode = func?.type;
         if (!returnTypeNode) {
             return undefined;
         }
 
         const returnType = checker.getTypeFromTypeNode(returnTypeNode);
         const promisedType = checker.getAwaitedType(returnType) || checker.getVoidType();
-        const promisedTypeNode = checker.typeToTypeNode(promisedType);
+        const promisedTypeNode = checker.typeToTypeNode(promisedType, /*enclosingDeclaration*/ func, /*flags*/ undefined);

```

Comment:
Why not use the `returnTypeNode` as the enclosing declaration (misnomer, is really the "name lookup location", don't worry about it not being a declaration), since it's present and should always have identical scoping to the target, since we're looking to replace it.

Reply:
Maybe we should rename it, because I made the exact same mistake.